### PR TITLE
feat(py_wheel): Add support for specifying Description-Content-Type and Summary in METADATA

### PR DESCRIPTION
**Pull request by sfc-gh-zpeng**
`py_wheel` allows to supply a description file, but it does not allow specifying the content type of that file.
By default, the type is "text/x-rst", but many packages are using a markdown description.

https://packaging.python.org/en/latest/specifications/core-metadata/#description-content-type

This change added the support.

### CODE REVIEW

**Comment by sfc-gh-zpeng** [comment]
@sfc-gh-sdas (fyi)

File: python/private/py_wheel.bzl [hunk 1]
```diff
@@ -275,6 +278,8 @@ def _py_wheel_impl(ctx):
         metadata_contents.append("Home-page: %s" % ctx.attr.homepage)
     if ctx.attr.license:
         metadata_contents.append("License: %s" % ctx.attr.license)
+    if ctx.attr.description_content_type:
```
Comment (rickeylev) [inline python/private/py_wheel.bzl:281]: How about setting the content type based on the file extension if it isn't specified? It looks like there's just .md, .rst, and .txt that need to be checked for.

**Review by rickeylev — CHANGES_REQUESTED** [review]

File: python/private/py_wheel.bzl (hunk 1 above)
Comment (sfc-gh-zpeng) [reply python/private/py_wheel.bzl:292]: Done. PTAL

**Comment by sfc-gh-zpeng** [comment]
I also added the "Summary" field: https://packaging.python.org/en/latest/specifications/core-metadata/#summary

File: python/private/py_wheel.bzl [hunk 2]
```diff
@@ -275,6 +289,20 @@ def _py_wheel_impl(ctx):
...
+        metadata_contents.append("Description-Content-Type: %s" % ctx.attr.description_content_type)
+    elif ctx.attr.description_file:
+        # infer the content type from description file extension.
+        description_file_type = _DEFAULT_DESCRIPTION_FILE_TYPE
+        description_files = ctx.attr.description_file.files.to_list()
+        if len(description_files) == 1:
+            description_file_type = _DESCRIPTION_FILE_EXTENSION_TO_TYPE.get(
+                description_files[0].extension,
+                _DEFAULT_DESCRIPTION_FILE_TYPE,
+            )
```
Comment (rickeylev) [inline python/private/py_wheel.bzl:302]:
```suggestion
        description_file_type = _DESCRIPTION_FILE_EXTENSION_TO_TYPE.get(
            ctx.file.description_file.extension,
            _DEFAULT_DESCRIPTION_FILE_TYPE,
        )
```

**Review by rickeylev — APPROVED** [review]

**Comment by rickeylev** [comment]
Docs look great, thank you!

**Pull request merged**